### DISCUSSION TITLE
rndint never returns the high value. No need for -1

### DISCUSSION THREAD

**Discussion by xia0** [ScureX/Takyon.PlayerVote#35] [General]
Has anyone noticed this? e.g. rndint(2) will always either return 0 or 1. I don't think -1 of the return value is indicated. If left how it currently is, surely the last map in the config will never be proposed.

**Comment by ScureX** [comment]
This... explains a lot. Yikes. So rndint(int max) will return a number from 0 to max-1?

**Reply by xia0** [reply]
From my testing, yes. I've not yet encountered an array out of bounds error. If it is indeed the same function documented here: https://developer.electricimp.com/examples/random the comment states the following:
`// Generate a pseudo-random integer between 0 and max - 1, inclusive`

**Reply by ScureX** [reply]
alr, will be fixed in 3.1.2! Thank you so much for pointing this out ive been confused for weeks lmao
